### DISCUSSION TITLE
Media options for media transformations

### DISCUSSION THREAD

**Discussion by Cthuluz** [tinacms/tinacms#5279] [Ideas]
I could be nice to have the option to have the media be transformed when uploaded. Perhaps there could be something in the defineConfig such as:

export default defineConfig({
...
  media: {
      tina: {
        mediaRoot: "",
        publicFolder: "static",
        static: false,
        mediaTransform: yourOwnTransformFunction,
      },
    }
...
});

**Comment by kldavis4** [comment]
@wicksipedia the tina cloud media implementation already does transformations (currently it creates thumbnails) so this should be technically possible

**Comment by wicksipedia** [comment]
Hey @Cthuluz 
I'd like to understand this a bit more. What types of transformations were you thinking about? Just resizing an image.... or to do something else as well?
